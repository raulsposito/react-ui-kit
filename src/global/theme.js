// https://www.color-hex.com/color/8293ee

export const colors = {
  primary: {
    regular: '#FC760D',
    dark: '#E46B0B',
    darker: '#C66F2B',
    hover: '#E17622',
    focus: '#B97641',
    selected: '#D4762C',
    disabled: '#F99546'
  },
  variant: {
    regular: '#FCB10D',
    dark: '#E29F0C',
    darker: '#C9A557',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  secondary: {
    regular: '#2FCFCF',
    dark: '#1F9A9A',
    darker: '#147676',
    hover: '#3DBDBD',
    focus: '#52A2A2',
    selected: '#44B4B4',
    disabled: '#5FD8D8'
  },
  success: {
    regular: '#3DB265',
    dark: '#48A668',
    darker: '#59946D',
    hover: '#48A668',
    focus: '#59946D',
    selected: '#59946D',
    disabled: '#9EBCA8'
  },
  error: {
    regular: '#FF5668',
    dark: '#E45C6B',
    darker: '#BA666F',
    hover: '#E45C6B',
    focus: '#BA666F',
    selected: '#BA666F',
    disabled: '#F8A4AC'
  },
  warning: {
    regular: '#FFAF00',
    dark: '#E5BB4D',
    darker: '#CCA644',
    hover: '#E4A418',
    focus: '#BA923B',
    selected: '#D69E23',
    disabled: '#FBBF3C'
  },
  contrast: {
    regular: '#3C3C3C',
    disabled: '#E4E4E4'
  },
  system: {
    regular: '#BAB5B4',
    dark: '#CBC6C4',
    darker: '#D7D4D1',
    hover: '#ACA8A8',
    focus: '#989595',
    selected: '#A6A2A1',
    disabled: '#C8C4C3'
  },
  background: {
    primary: '#FFFFFF',
    regular: '#FFFFFF',
    hover: '#F1F1F1',
    disabled: '#F8F8F8'
  },
  events: {
    inprogress: '#3DB265',
    next: '#5678CF',
    finished: '#BAB5B4',
    suspended: '#FFD056',
    cancelled: '#FF5668',
    white: '#FFFFFF'
  }
}

export const spacing = {
  quarter: '2px',
  half: '4px',
  threeQuarters: '6px',
  one: '8px',
  oneAndAQuarter: '10px',
  oneAndAHalf: '12px',
  two: '16px',
  twoAndAQuarter: '18px',
  twoAndAHalf: '20px',
  three: '24px',
  threeAndAHalf: '28px',
  four: '32px',
  fourAndAHalf: '36px',
  five: '40px',
  fiveAndAHalf: '44px',
  six: '48px',
  sixAndAHalf: '52px',
  seven: '56px',
  sevenAndAHalf: '60px',
  eight: '64px',
  eightAndAHalf: '68px',
  nine: '72px',
  nineAndAHalf: '76px',
  ten: '80px',
  tenAndAHalf: '84px',
  eleven: '88px',
  elevenAndAHalf: '92px',
  twelve: '96px',
  twelveAndAHalf: '100px',
  thirteen: '104px',
  fourteen: '112x',
  fifteen: '120px'
}

export const page = {
  width: '1200px'
}
