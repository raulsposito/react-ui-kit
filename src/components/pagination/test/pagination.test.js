/* eslint-disable */
import React from 'react'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { colors, spacing } from '@global/theme'

import Pagination from '..'

const mock = {
  pages: 13,
  amount: 7,
  goToPage: jest.fn(),
  selected: 3,
  showStartAndEnd: true,
  showBackAndNext: true
}

describe('<Pagination />', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Pagination {...mock} />)
    expect(asFragment()).toMatchSnapshot()
  })

  describe('actions', () => {
    describe('arrows', () => {
      it('FIRST -> goes to first element', () => {
        render(<Pagination {...mock} />)
        const toStart = screen.getAllByRole('button')[0]
        userEvent.click(toStart)
        expect(mock.goToPage).toHaveBeenCalledWith(0)
      })

      it('PREVIOUS -> goes to previeous if not first', () => {
        render(<Pagination {...mock} />)
        const prev = screen.getAllByRole('button')[1]
        userEvent.click(prev)
        expect(mock.goToPage).toHaveBeenCalledWith(mock.selected - 1)
      })

      it('LAST -> goes to last element', () => {
        render(<Pagination {...mock} />)
        const [toEnd] = screen.getAllByRole('button').slice(-1)
        userEvent.click(toEnd)
        expect(mock.goToPage).toHaveBeenCalledWith(mock.pages - 1)
      })

      it('NEXT -> goes to next if not last', () => {
        render(<Pagination {...mock} />)
        const [next] = screen.getAllByRole('button').slice(-2)
        userEvent.click(next)
        expect(mock.goToPage).toHaveBeenCalledWith(mock.selected + 1)
      })
    })

    it('goes to page when clicked on the number', () => {
      render(<Pagination {...mock} selected={6} />)
      const num = screen.getAllByRole('button')[4]
      userEvent.click(num)
      expect(mock.goToPage).toHaveBeenCalledWith(5)
    })
  })

  describe('renders arrows', () => {
    it('has first and last arrows', () => {
      render(<Pagination {...mock} />)
      expect(screen.queryByText('first_page')).toBeInTheDocument()
      expect(screen.queryByText('last_page')).toBeInTheDocument()
    })

    it('does not has first and last arrows', () => {
      render(<Pagination {...mock} showStartAndEnd={false} />)
      expect(screen.queryByText('first_page')).not.toBeInTheDocument()
      expect(screen.queryByText('last_page')).not.toBeInTheDocument()
    })

    it('has back and next arrows', () => {
      render(<Pagination {...mock} />)
      expect(screen.queryByText('chevron_left')).toBeInTheDocument()
      expect(screen.queryByText('chevron_right')).toBeInTheDocument()
    })

    it('does not has back and next arrows', () => {
      render(<Pagination {...mock} showBackAndNext={false} />)
      expect(screen.queryByText('chevron_left')).not.toBeInTheDocument()
      expect(screen.queryByText('chevron_right')).not.toBeInTheDocument()
    })
  })

  describe('renders numbers and elipsis', () => {
    it('has less pages than amount', () => {
      render(<Pagination {...mock} pages={5} />)
      expect(screen.queryByTestId('more_horiz')).not.toBeInTheDocument()
    })

    describe('more pages than amount', () => {
      it('shows elipsis at the end', () => {
        render(<Pagination {...mock} />)
        const ellipsis = screen.queryAllByText('more_horiz')
        expect(ellipsis.length).toEqual(1)

        const [ellipse] = screen.queryAllByRole('button').slice(-4)
        expect(ellipse.textContent).toEqual('more_horiz')
      })

      it('shows elipsis at the start', () => {
        render(<Pagination {...mock} selected={12} />)
        const ellipsis = screen.queryAllByText('more_horiz')
        expect(ellipsis.length).toEqual(1)

        const ellipse = screen.queryAllByRole('button')[3]
        expect(ellipse.textContent).toEqual('more_horiz')
      })

      it('shows elipsis at both sides', () => {
        render(<Pagination {...mock} selected={7} />)
        const ellipsis = screen.queryAllByText('more_horiz')
        expect(ellipsis.length).toEqual(2)

        const buttons = screen.queryAllByRole('button')
        expect(buttons[3].textContent).toEqual('more_horiz')
        expect(buttons[7].textContent).toEqual('more_horiz')
      })
    })
  })

  describe('size', () => {
    it('has default size', () => {
      render(<Pagination {...mock} />)
      const buttons = screen.queryAllByRole('button')
      buttons.forEach(btn => {
        expect(btn).toHaveStyle({
          width: `${spacing.fourAndAHalf}`,
          height: `${spacing.fourAndAHalf}`
        })
      })
    })

    it('has default small', () => {
      render(<Pagination {...mock} size='small' />)
      const buttons = screen.queryAllByRole('button')
      buttons.forEach(btn => {
        expect(btn).toHaveStyle({
          width: `${spacing.threeAndAHalf}`,
          height: `${spacing.threeAndAHalf}`
        })
      })
    })

    it('has default medium', () => {
      render(<Pagination {...mock} size='medium' />)
      const buttons = screen.queryAllByRole('button')
      buttons.forEach(btn => {
        expect(btn).toHaveStyle({
          width: `${spacing.fourAndAHalf}`,
          height: `${spacing.fourAndAHalf}`
        })
      })
    })

    it('has default large', () => {
      render(<Pagination {...mock} size='large' />)
      const buttons = screen.queryAllByRole('button')
      buttons.forEach(btn => {
        expect(btn).toHaveStyle({
          width: `${spacing.five}`,
          height: `${spacing.five}`
        })
      })
    })
  })

  describe('color', () => {
    it('has default background color', () => {
      render(<Pagination {...mock} />)
      const btn = screen.queryByText(`${mock.selected + 1}`).parentNode
      expect(btn).toHaveStyle({ 'background-color': `${colors.primary.selected}` })
    })

    it('has background color', () => {
      render(<Pagination {...mock} color='variant' />)
      const btn = screen.queryByText(`${mock.selected + 1}`).parentNode
      expect(btn).toHaveStyle({ 'background-color': `${colors.variant.selected}` })
    })
  })

  describe('disabled', () => {
    it('does not call goToPage', () => {
      render(<Pagination {...mock} disabled />)
      const btn = screen.queryAllByRole('button')[0]
      userEvent.click(btn)
      expect(mock.goToPage).not.toHaveBeenCalled()
    })

    it('has disabled color', () => {
      render(<Pagination {...mock} disabled />)
      const arrow = screen.queryByText('first_page')
      expect(arrow).toHaveStyle({color: `${colors.system.disabled}`})

      const one = screen.queryByText('1')
      expect(one).toHaveStyle({color: `${colors.system.disabled}`})
    })
  })
})