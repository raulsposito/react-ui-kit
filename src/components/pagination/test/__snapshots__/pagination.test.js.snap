// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-kEjbQP itiNpX"
  >
    <button
      class="sc-jrAFXE jqYNa-D"
    >
      <span
        class="sc-iBPTik bAqRBC material-icons"
        color="contrast"
      >
        first_page
      </span>
    </button>
    <button
      class="sc-jrAFXE jqYNa-D"
    >
      <span
        class="sc-iBPTik bAqRBC material-icons"
        color="contrast"
      >
        chevron_left
      </span>
    </button>
    <button
      class="sc-jrAFXE jqYNa-D"
    >
      <p
        class="sc-bdfBQB iDqXfG"
        color="contrast"
      >
        1
      </p>
    </button>
    <button
      class="sc-jrAFXE jqYNa-D"
    >
      <p
        class="sc-bdfBQB iDqXfG"
        color="contrast"
      >
        2
      </p>
    </button>
    <button
      class="sc-jrAFXE jqYNa-D"
    >
      <p
        class="sc-bdfBQB iDqXfG"
        color="contrast"
      >
        3
      </p>
    </button>
    <button
      class="sc-jrAFXE czXxsN"
    >
      <p
        class="sc-bdfBQB iDqXfG"
        color="contrast"
      >
        4
      </p>
    </button>
    <button
      class="sc-jrAFXE jqYNa-D"
    >
      <p
        class="sc-bdfBQB iDqXfG"
        color="contrast"
      >
        5
      </p>
    </button>
    <button
      class="sc-jrAFXE txkfq"
      color="contrast"
    >
      <span
        class="sc-iBPTik bAqRBC material-icons"
        color="contrast"
      >
        more_horiz
      </span>
    </button>
    <button
      class="sc-jrAFXE jqYNa-D"
    >
      <p
        class="sc-bdfBQB iDqXfG"
        color="contrast"
      >
        13
      </p>
    </button>
    <button
      class="sc-jrAFXE jqYNa-D"
    >
      <span
        class="sc-iBPTik bAqRBC material-icons"
        color="contrast"
      >
        chevron_right
      </span>
    </button>
    <button
      class="sc-jrAFXE jqYNa-D"
    >
      <span
        class="sc-iBPTik bAqRBC material-icons"
        color="contrast"
      >
        last_page
      </span>
    </button>
  </div>
</DocumentFragment>
`;
